OK, this should be good to go. Thanks, Thomas for the suggestion to run a "test release."

This way, I uncovered an issue with our new file-limits support (https://github.com/lycheeverse/lychee/pull/2106). 

On Windows, I got

```rust
error[E0432]: unresolved imports `rlimit::Resource`, `rlimit::getrlimit`
   --> lychee-bin\src\main.rs:143:18
    |
143 |     use rlimit::{Resource, getrlimit, increase_nofile_limit};
    |                  ^^^^^^^^  ^^^^^^^^^ no `getrlimit` in the root
    |                  |
    |                  no `Resource` in the root
For more information about this error, try `rustc --explain E0432`.
error: could not compile `lychee` (bin "lychee") due to 1 previous error
Caused by:
  process didn't exit successfully:
```

Inside the `rlimit` crate, `increase_nofile_limit(u64::MAX)` evaluates internally via `#[cfg(unix)]` to attempt bumping to the max hard limit. If it fails, it still gracefully returns `Ok(current_soft_limit)`. And on Windows, via `#[cfg(windows)]`, it simply returns `Ok(lim)`. Source code [here](https://github.com/Nugine/rlimit/blob/94c9e171e9d11690756cb15dc7dac07ad9067478/src/tools.rs#L44).

This means `rlimit` is already entirely cross-platform and already provides the *exact fallback semantics* we want internally! The entire `limit_result` match statement with `cfg(unix)` and `cfg(not(unix))` was just redundant boilerplate. :)

This run shows that all binaries build successfully: https://github.com/lycheeverse/lychee/actions/runs/23962659010/job/69895885187

actions/upload-release-asset@v1 is deprecated and relies on older, outdated versions of Node.js. The `gh` CLI tool is natively pre-installed on all GitHub-hosted runners (Ubuntu, macOS, and Windows). This makes it universally available without needing to download external dependencies or rely on third-party actions.

--- a/.github/workflows/release-binary.yml
+++ b/.github/workflows/release-binary.yml
@@ -210,7 +210,7 @@ jobs:
           mkdir -p "$ARCHIVE/docs" "$ARCHIVE/complete"
           cp "$BIN" "$ARCHIVE"/
           cp README.md "$ARCHIVE"/
-          cp docs/*.md "$ARCHIVE/docs/" || true
+          cp docs/*.md "$ARCHIVE/docs/"
 
       - name: Generate man page (no emulation)
         if: matrix.qemu == ''

